### PR TITLE
2640 queued agent event publisher

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #2640 .

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by unit tests
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/common/event_queue/queued_agent_event_publisher.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+from common.agent_events import AbstractAgentEvent
+
+from . import IAgentEventPublisher
+
+
+class QueuedAgentEventPublisher(IAgentEventPublisher):
+    def __init__(self, queue: Queue):
+        self._queue = queue
+
+    def publish(self, event: AbstractAgentEvent):
```
Comment (VakarisZ) [inline monkey/common/event_queue/queued_agent_event_publisher.py:12]: Shouldn't there be a getter for the `self._queue`? Or will that be part of another PR?
